rest_api/sawtooth_rest_api/route_handlers.py

```diff
@@ -505,17 +511,27 @@ def _check_status_errors(proto, content, error_traps=None):
                 trap.check(content.status)
 
     @staticmethod
-    def _wrap_response(data=None, metadata=None, status=200):
+    def _wrap_response(request, data=None, metadata=None, status=200):
         """Creates the JSON response envelope to be sent back to the client.
         """
         envelope = metadata or {}
 
         if data is not None:
             envelope['data'] = data
 
+        headers = None
+        if 'Origin' in request.headers:
+            headers = {
+                'Access-Control-Allow-Origin': request.headers['Origin'],
+                "Access-Control-Allow-Methods": "GET,POST",
```

Comment:
This could be refactored into a method on the RouteHandler, or maybe a separate utility class, and then called here and in `rest_api.py`.